plextraktsync/plex_server.py

```diff
@@ -1,92 +1,110 @@
 from functools import partial
+from typing import List
 
 import plexapi.server
 
 from plextraktsync.config import PLEX_PLATFORM
 from plextraktsync.decorators.nocache import nocache
-from plextraktsync.factory import factory
+from plextraktsync.factory import Factory
 from plextraktsync.logging import logger
 
 
 class PlexServerConnection:
-    @staticmethod
+    def __init__(self, factory: Factory):
+        self.factory = factory
+
+    @property
+    def timeout(self):
+        return self.config["plex"]["timeout"]
+
+    @property
+    def config(self):
+        return self.factory.config()
+
+    @property
+    def session(self):
+        return self.factory.session()
+
     @nocache
-    def connect():
-        return _get_plex_server()
+    def connect(self, urls: List[str], token: str):
+        plexapi.X_PLEX_PLATFORM = PLEX_PLATFORM
+        plexapi.TIMEOUT = self.timeout
+        plexapi.BASE_HEADERS["X-Plex-Platform"] = plexapi.X_PLEX_PLATFORM
+
+        PlexServer = partial(plexapi.server.PlexServer, session=self.session)
+
+        # if connection fails, it will try:
+        # 1. url expected by new ssl certificate
+        # 2. url without ssl
+        # 3. local url (localhost)
+        # for url in walk():
+        for url in urls:
+            try:
+                return PlexServer(token=token, baseurl=url)
+            except plexapi.server.requests.exceptions.SSLError as e:
+                logger.error(e)
+                message = str(e.__context__)
+
+                # 1.
+                # HTTPSConnectionPool(host='127.0.0.1', port=32400):
+                # Max retries exceeded with url: / (
+                #  Caused by SSLError(
+                #   CertificateError(
+                #     "hostname '127.0.0.1' doesn't match '*.5125cc430e5f1919c27226507eab90df.plex.direct'"
+                #    )
+                #  )
+                # )
+                if "doesn't match '*." in message and ".plex.direct" in url:
+                    url = self.extract_plex_direct(url, message)
+                    logger.warning(f"Trying with url: {url}")
+                    urls.append(url)
+                    self.save_new_url(url, urls[-1])
```

Comment:
I don't like this saving over the new config here. doesn't really fit with what the method is doing here. I think it's fine to do that every time the error is thrown.